components/transactions/TransactionItem.vue

```diff
@@ -109,16 +141,71 @@ export default {
           return `Vote`
         case lunieMessageTypes.CLAIM_REWARDS:
           return `Claim Rewards`
+        case lunieMessageTypes.CREATE_ISCN_RECORD:
+          return `Create ISCN Record`
+        case lunieMessageTypes.UPDATE_ISCN_RECORD:
+          return `Update ISCN Record`
+        case lunieMessageTypes.CHANGE_ISCN_OWNERSHIP:
+          return `Change ISCN Ownership`
         case lunieMessageTypes.UNKNOWN:
-          return this.transaction.rawMessage.type.split('/Msg')[1]
+          return this.transaction.rawMessage.message['@type'].split('/')[1]
         /* istanbul ignore next */
         default:
           return ``
       }
     },
+    iscnId() {
+      if (
+        this.transaction.type === lunieMessageTypes.CREATE_ISCN_RECORD ||
+        this.transaction.type === lunieMessageTypes.UPDATE_ISCN_RECORD ||
+        this.transaction.type === lunieMessageTypes.CHANGE_ISCN_OWNERSHIP
+      ) {
+        let iscnId = ''
+        for (let i = 0; i < this.transaction.events.length; i++) {
+          if (this.transaction.events[i]) {
+            for (let j = 0; j < this.transaction.events[i].length; j++) {
+              for (let k = 0; k < this.transaction.events[i][j].attributes.length; k++) { // eslint-disable-line
+                if (this.transaction.events[i][j].attributes[k].key === 'iscn_id') { // eslint-disable-line
+                  iscnId = this.transaction.events[i][j].attributes[k].value
+                  return iscnId
+                }
+              }
```

Comment:
Can we use .find() for this final loop?

Reply:
Also for `// eslint-disable-line`, please explicit disable a rule instead of the whole eslint